testsuite/tests/parsetree/source_jane_street.ml

```diff
@@ -1005,10 +1005,10 @@ module Value1 : sig end
 module Value2 : sig end -> sig end -> sig end
 module Name2_1 : sig end
 module Name2_1 : sig end
->> Fatal error: Unimplemented: instance identifier
-                Base[Name1:Value1][Name2:Value2[Name2_1:Value2_1]]
-Uncaught exception: Misc.Fatal_error
-
+Line 9, characters 11-64:
+9 | module _ = Base(Name1)(Value1)(Name2)(Value2(Name2_1)(Value2_1)) [@jane.non_erasable.instances]
+               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+Error: Unbound module "Base[Name1:Value1][Name2:Value2[Name2_1:Value2_1]]"
```

Comment:
Might be worthwhile to make the printed instance syntax same as the user-syntax? Maybe currently end-users might never see this, but still.

Reply:
I don't think it's worth writing the code to format it in a way that we don't intend for humans to ever actually write. It does make the error look a bit funny now that we have location information, but I don't think it helps to make it look hideous both times

Reply:
Yeah I don't have strong opinion on this. Let's just resolve.